admin/class-monk-admin.php

```diff
@@ -1294,4 +1293,59 @@ public function monk_change_nav_menu_fields() {
 		}
 		require_once plugin_dir_path( __FILE__ ) . '/partials/admin-monk-select-menu-to-edit-render.php';
 	}
+
+	/**
+	 * Download language packages according to active languages
+	 *
+	 * Update active and default languages.
+	 *
+	 * @since    0.4.0
+	 *
+	 * @return void
+	 */
+	public function monk_save_language_packages() {
+		if ( check_ajax_referer( '_monk_nonce', '_monk_nonce', false ) ) {
+			global $monk_languages;
+			$active_languages  = $_POST['monk_active_languages'];
+			$default_language  = $_POST['monk_default_language'];
+
+			require_once( ABSPATH . 'wp-admin/includes/translation-install.php' );
+
+			$success      = __( '<strong>This packages have been downloaded:</strong>', 'monk' ) . '<br />';
+			$have_success = false;
+			$error        = __( '<strong>This packages have not been downloaded:</strong>', 'monk' ) . '<br />';
+			$have_error   = false;
+
+			foreach ( $active_languages as $language ) {
+				if ( 'en_US' !== $language ) {
+					$download_pack = wp_download_language_pack( $language );
+					if ( $download_pack ) {
+						$success      .= $monk_languages[ $download_pack ]['name'] . '<br />';
+						$have_success = true;
+					} else {
+						$error      .= $monk_languages[ $language ]['name'] . '<br />';
+						$have_error = true;
+					}
+				}
+			}
+
+			$response = '';
+			if ( $have_success ) {
+				$response .= '<div class="notice notice-success is-dismissible"><p>' . $success
+				. '</p></div>';
+			}
+
+			if ( $have_error ) {
+				$response .= '<div class="notice notice-error is-dismissible"><p>' . $error . '</p></div>';
+			}
+
+			update_option( 'monk_active_languages', $active_languages );
```

Comment:
Remove these updates. Just download the language packages.
Let Settings API do its job.